### ISSUE TITLE
Uncaught Error: Connection is closed.

### ISSUE THREAD

**Issue by liumenglz** [atom/ide-typescript#118]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.28.2 x64
**Electron**: 2.0.5
**OS**: Unknown Windows version
**Thrown From**: [ide-typescript](https://github.com/atom/ide-typescript) package 0.7.6

### Stack Trace

Uncaught Error: Connection is closed.

```
At C:\Users\Meng\.atom\packages\ide-typescript\node_modules\vscode-jsonrpc\lib\main.js:613

Error: Connection is closed.
    at /packages/ide-typescript/node_modules/vscode-jsonrpc/lib/main.js:138:28)
    at throwIfClosedOrDisposed (/packages/ide-typescript/node_modules/vscode-jsonrpc/lib/main.js:613:19)
    at Object.sendNotification (/packages/ide-typescript/node_modules/vscode-jsonrpc/lib/main.js:667:13)
    at LanguageClientConnection._sendNotification (/packages/ide-typescript/node_modules/atom-languageclient/build/lib/languageclient.js:341:19)
    at LanguageClientConnection.didChangeTextDocument (/packages/ide-typescript/node_modules/atom-languageclient/build/lib/languageclient.js:133:14)
    at TextEditorSyncAdapter.sendFullChanges (/packages/ide-typescript/node_modules/atom-languageclient/build/lib/adapters/document-sync-adapter.js:158:26)
    at Function.module.exports.Emitter.simpleDispatch (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/event-kit/lib/emitter.js:27:20)
    at Emitter.module.exports.Emitter.emit (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/event-kit/lib/emitter.js:156:34)
    at disposables.add.displayLayer.onDidChange.changes (~/AppData/Local/atom/app-1.28.2/resources/app/src/text-editor.js:614:26)
    at Function.module.exports.Emitter.simpleDispatch (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/event-kit/lib/emitter.js:27:20)
    at Emitter.module.exports.Emitter.emit (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/event-kit/lib/emitter.js:156:34)
    at DisplayLayer.emitDeferredChangeEvents (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/text-buffer/lib/display-layer.js:819:26)
    at TextBuffer.module.exports.TextBuffer.emitDidChangeTextEvent (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/text-buffer/lib/text-buffer.js:1800:30)
    at TextBuffer.module.exports.TextBuffer.transact (~/AppData/Local/atom/app-1.28.2/resources/app/node_modules/text-buffer/lib/text-buffer.js:996:18)
    at TextEditor.transact (~/AppData/Local/atom/app-1.28.2/resources/app/src/text-editor.js:2002:30)
    at TextEditor.moveCursors (~/AppData/Local/atom/app-1.28.2/resources/app/src/text-editor.js:2875:23)
    at TextEditor.setCursorScreenPosition (~/AppData/Local/atom/app-1.28.2/resources/app/src/text-editor.js:2686:23)
    at TextEditorComponent.didMouseDownOnContent (~/AppData/Local/atom/app-1.28.2/resources/app/src/text-editor-component.js:1808:25)
```

### Commands

```
     -4:21.1.0 core:paste (input.hidden-input)
     -4:17.6.0 intentions:highlight (input.hidden-input)
     -4:17.5.0 core:save (input.hidden-input)
     -4:12.2.0 intentions:highlight (input.hidden-input)
     -4:12.1.0 core:save (input.hidden-input)
     -4:10.4.0 intentions:highlight (input.hidden-input)
     -4:10.2.0 core:save (input.hidden-input)
     -4:07.1.0 core:backspace (input.hidden-input)
     -3:41.7.0 intentions:highlight (input.hidden-input)
  5x -3:41.5.0 core:save (input.hidden-input)
     -3:38.4.0 intentions:highlight (input.hidden-input)
     -3:38.3.0 core:save (input.hidden-input)
     -2:58.2.0 intentions:highlight (input.hidden-input)
 11x -2:58.1.0 core:undo (input.hidden-input)
     -2:51.5.0 intentions:highlight (input.hidden-input)
  2x -2:51.4.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-beautify 0.32.2 
atom-ide-ui 0.7.2 
atom-typescript 12.6.2 
autoclose-html 0.23.0 
busy-signal 1.4.3 
file-icons 2.1.19 
highlight-selected 0.13.1 
ide-typescript 0.7.6 
intentions 1.1.5 
linter 2.2.0 
linter-eslint 8.4.1 
linter-tslint 1.9.1 
linter-ui-default 1.7.1 
minimap 4.29.8 
open-recent 5.0.0 
pigments 0.40.2 
regex-railroad-diagram 0.19.4 
```

**Comment by rsese** [comment]
Thanks for the report - is the error reproducible for you?  Auto-restarting seems to have helped with previous reports of this error x-ref: https://github.com/atom/ide-typescript/issues/17#issuecomment-366087773.

**Comment by damieng** [comment]
Should be fixed with the new ide-typescript 0.8.0 update.
